### ISSUE TITLE
Online Help

### ISSUE THREAD

**Issue by fwilhe** [osiam/osiam#86]
Hi all,

I've taken a look at `asciidoctor` for writing documentation. It integrates nicely with the java build tools (maven and gradle). Furthermore I found [this script](https://github.com/fkling/esprima_ast_explorer/blob/master/scripts/push-site.sh) in a completely unrelated project, but I think we could make use of that to publish the generated html documentation via gh pages. I adopted it for our use-case ([gist](https://gist.github.com/fwilhe/d0cb0a954d902c892856)), and tried this with my fork of the shell. It worked quite nicely.

My question is: What do you think about that in general?

There is potentially more documentation that we could publish that way, for example the API documentation (#75) or java doc, if that makes sense.

**Comment by wallner** [comment]
I like the solution you sketched out very much. Although I see the problem that someone still has to _write_ the documentation. So, as long as we don't put the nose to the grind stone and write some documents, the best process for documentation publication is basically worthless, because we still don't have anything to publish.

**Comment by fwilhe** [comment]
> someone still has to write the documentation

That is absolutely correct. I am interested in improving my writing skills, so I am willing to take that task. I might need help at some points where my knowledge is not yet deep enough, but I don't see a problem there.

**Comment by tkrille** [comment]
Looks really nice. If you have questions about OSIAM, that you need an answer for to write docs, please open a new issue and ask them.
